src/passes/TypeMerging.cpp

```diff
@@ -591,6 +591,15 @@ bool shapeEq(HeapType a, HeapType b) {
   if (a.isShared() != b.isShared()) {
     return false;
   }
+  if (!!a.getDeclaredSuperType() != !!b.getDeclaredSuperType()) {
+    return false;
+  }
```

Comment:
The supertype is not part of CD - was this a bug all along?

Do we need a test for that too?

Reply:
I'm not 100% sure this part of the change was necessary, but I'm 100% sure it's safe and correct. I'll spend a minute or two trying to see if I can come up with a test that demonstrates a fixed bug.